query/extract.js

```diff
@@ -1,50 +1,68 @@
-
 // maximum names to match on
-var MAX_NAMES = 10;
+const MAX_NAMES = 10;
 
 // maximum address records to return
-var MAX_MATCHES = 5000; // note: this query should only be used for debugging purposes
+const MAX_MATCHES = 5000; // note: this query should only be used for debugging purposes
+
+const SQL = `
+  SELECT address.* FROM street.rtree
+  JOIN street.names ON street.names.id = street.rtree.id
+  JOIN address ON address.id = street.rtree.id
+  WHERE (
+    street.rtree.minX<=$lon AND street.rtree.maxX>=$lon AND
+    street.rtree.minY<=$lat AND street.rtree.maxY>=$lat
+  )
+  AND ( %%NAME_CONDITIONS%% )
+  ORDER BY address.housenumber ASC // @warning business logic depends on this
+  LIMIT ${MAX_MATCHES};
+`;
 
-var SQL = [
-  'SELECT address.* FROM street.rtree',
-  'JOIN street.names ON street.names.id = street.rtree.id',
-  'JOIN address ON address.id = street.rtree.id',
-  'WHERE (',
-    'street.rtree.minX<=?1 AND street.rtree.maxX>=?1 AND',
-    'street.rtree.minY<=?2 AND street.rtree.maxY>=?2',
-  ')',
-  'AND ( %%NAME_CONDITIONS%% )',
-  'ORDER BY address.housenumber ASC', // @warning business logic depends on this
-  'LIMIT %%MAX_MATCHES%%;'
-].join(' ');
+// SQL prepared statements dont easily support variable length inputs.
+// This function dynamically generates a SQL query based on the number
+// of 'name' conditions required.
+function generateDynamicSQL(max) {
```

Comment:
this function is duplicated three times, as is the comment, why not make it a helper?